subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperGenerationIntegrationTest.groovy

```diff
@@ -59,6 +60,22 @@ class WrapperGenerationIntegrationTest extends AbstractIntegrationSpec {
         file("gradle/wrapper/gradle-wrapper.properties").text.contains("distributionUrl=https\\://services.gradle.org/distributions/gradle-2.2.1-bin.zip")
     }
 
+    def "generated wrapper scripts for valid distribution type from command-line"() {
```

Comment:
Are you suggesting to not add a separate test, but use a combined `bin` and `all` tests? Also, as I'm not very familiar with Spock, could you please elaborate in more detail how to make use of `where` here?